### PR TITLE
Add LIP BLS signatures

### PR DESCRIPTION
**Pull request by AndreasKendziorra**

### CODE REVIEW

File: proposals/lip-BLS_signatures.md [hunk 1]
```diff
@@ -0,0 +1,171 @@
...
+
+#### Keypair Creation
+
+A secret key is created using `KeyGen`. The input for `KeyGen` must be an infeasible to guess octet string of length at least 32. See the [appendix](#key-management) for a recommendation on how to choose this input and on key management.
+
+The public key for a secret key `sk` is created by `SkToPk(sk)`.
+
+#### Signing and Verifying
+
+Let `m` be a binary message, `tag` the correct message tag for `m` as specified in  LIP [0037][lip-use-message-tags-and-network-identifiers-for-signatures], `networkIdentifier` the correct network identifier of the chain and `sk` a secret key. Then, the signature is computed by `signBLS(sk, tag, networkIdentifier, m)` as defined below. The resulting signature `sig` in combination with the message `m` and the matching public key `pk` is verified by `verifyBLS(pk, tag, networkIdentifier, m, sig)`. In the following, let `tagMessage` be the function defined in LIP [0037][lip-use-message-tags-and-network-identifiers-for-signatures].
```
Comment (MaximeGagnebin) [inline proposals/lip-BLS_signatures.md:57]:
```suggestion
Let `m` be a binary message, `tag` the correct message tag for `m` as specified in [LIP 0037][lip-use-message-tags-and-network-identifiers-for-signatures], `networkIdentifier` the correct network identifier of the chain and `sk` a secret key. Then, the signature is computed by `signBLS(sk, tag, networkIdentifier, m)` as defined below. The resulting signature `sig` in combination with the message `m` and the matching public key `pk` is verified by `verifyBLS(pk, tag, networkIdentifier, m, sig)`. In the following, let `tagMessage` be the function defined in [LIP 0037][lip-use-message-tags-and-network-identifiers-for-signatures].
```

**Review by MaximeGagnebin — APPROVED** [review]
Left only a small formatting comment. Looks good :)

File: proposals/lip-BLS_signatures.md (hunk 1 above)
Comment (AndreasKendziorra) [reply proposals/lip-BLS_signatures.md]: done

File: proposals/lip-BLS_signatures.md [hunk 2]
```diff
@@ -0,0 +1,171 @@
...
+
+The Lisk protocol could contain several transaction types that perform such a registration. In particular, there could be different registration transactions for different keys, e.g, one for validator public keys and one for public keys of regular accounts. This LIP does not specify any registration transactions. Such transaction types must be defined in separate LIPs. In the following, we just assume there exists such a transaction type which we call _register public key transaction_. To register the public key of the key pair `(sk, pk)` by a _register public key transaction_, `registerPublicKeyTransaction`, the transaction must contain `pk` and a proof, `prf`, generated by `PopProve(sk)`. If `prf` does not satisfy `PopVerify(pk, prf) == VALID`, then `registerPublicKeyTransaction` is invalid and must be rejected. Once `registerPublicKeyTransaction` is included, transactions and blocks that require to have a valid signature for `pk` can be included in the blockchain.
```
[the remark this answers is not in the document]
Comment (AndreasKendziorra) [reply proposals/lip-BLS_signatures.md:73]: I really want to distinguish here between the transaction type, _register public key transaction_, and the concrete instance of this transaction type, `registerPublicKeyTransaction`.

File: proposals/lip-BLS_signatures.md [hunk 3]
```diff
@@ -0,0 +1,171 @@
...
+    for every i in [0, …, ceil(length(keyList)/8)]:
+        if i-th bit of aggregationBits is set to 1:
+            keys.append(keysList[i])
+    return FastAggregateVerify(keys, taggedMessage, signature)
+```
+
+If one wants to additionally validate that the participating public keys satisfy a certain weight threshold, the function `verifyWeightedAggSig` can be used. The function takes additionally a list of weights, `weights`, where the i-th entry specifies the weight for the i-th public key in `keysList` and a weight threshold `threshold`.
+
+```python
+verifyWeightedAggSig(keysList, aggregationBits, signature, tag, networkIdentifier, m, weights, threshold):
```
[the remark this answers is not in the document]
Comment (AndreasKendziorra) [reply proposals/lip-BLS_signatures.md]: done

**Review by ricott1 — APPROVED** [review]
LGTM

**Review by ManuGowda — APPROVED** [review]

**Review by karmacoma — APPROVED** [review]

**Pull request merged**